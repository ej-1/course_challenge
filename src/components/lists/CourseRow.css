.selected {
  background: #5ca2b2;
  color: #fff; }
