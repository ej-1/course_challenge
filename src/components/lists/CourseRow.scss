@import "./main.scss";

.selected {
  background: $primary-background;
  color: $primary-font-color;
}
