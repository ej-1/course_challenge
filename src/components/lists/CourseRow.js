import React from "react";
import "../lists/CourseRow.css";

const CourseRow = ({ courseData, slug, onSelect }) => (
  <tr key={courseData.title} id={slug} onClick={() => onSelect(slug)}>
    <td>{courseData.title}</td>
    <td>{courseData.author}</td>
    <td>{courseData.next_start_formatted}</td>
  </tr>
);

export default CourseRow;
