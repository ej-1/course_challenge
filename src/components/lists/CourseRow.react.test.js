import React from "react";
import CourseRow from "./CourseRow";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

const onSelect = jest.fn();
// dont put in global scope
const courseData = {
  title: "How to write tests",
  author: "Erik W. Jonsson",
  next_start_formatted: "Monday, September 17th, 2018"
};
const slug = "/how-to-write-tests";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <CourseRow onSelect={onSelect} courseData={courseData} slug={slug} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("onSelect sends slug", () => {
  const wrapper = shallow(
    <CourseRow onSelect={onSelect} courseData={courseData} slug={slug} />
  );
  wrapper.find("tr").simulate("click");

  expect(onSelect).toHaveBeenCalledWith("/how-to-write-tests");
});
