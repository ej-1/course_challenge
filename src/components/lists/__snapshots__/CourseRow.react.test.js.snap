// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  id="/how-to-write-tests"
  onClick={[Function]}
>
  <td>
    How to write tests
  </td>
  <td>
    Erik W. Jonsson
  </td>
  <td>
    Monday, September 17th, 2018
  </td>
</tr>
`;
